apps/dot_ext/views/authorization.py

```diff
@@ -291,12 +295,42 @@ def dispatch(self, request, uuid, *args, **kwargs):
 class TokenView(DotTokenView):
     @method_decorator(sensitive_post_parameters("password"))
     def post(self, request, *args, **kwargs):
+        """Subclass of oauth2_provider.views.base.TokenView
+        to support checking application active/DAG status and
+        modification of the access token to include access
+        grant expiry.
+
+        RETURN:
+            response: HTTPResponse containing serialized access
+            token response.
+        """
         try:
-            validate_app_is_active(request)
+            dag = validate_app_is_active(request)[1]
+            if dag is not None and dag.expiration_date is not None:
+                dag_expiry = strftime('%Y-%m-%d %H:%M:%S', dag.expiration_date.timetuple())
+            else:
+                dag_expiry = None
         except (InvalidClientError, InvalidGrantError) as error:
             return json_response_from_oauth2_error(error)
 
-        return super().post(request, args, kwargs)
+        url, headers, body, status = self.create_token_response(request)
+        if status == 200:
+            body = json.loads(body)
+            access_token = body.get("access_token")
+            if access_token is not None:
+                token = get_access_token_model().objects.get(
+                    token=access_token)
+                app_authorized.send(
+                    sender=self, request=request,
+                    token=token)
+                if dag_expiry is None:
+                    dag_expiry = strftime('%Y-%m-%d %H:%M:%S', localtime(body.get("expires_at")))
```

Comment:
`expires_at` seems to be the expiration of the token, not the underlying dag, so we'll need to get this from elsewhere. In my testing I was seeing `'access_grant_expiration': '2024-06-13 21:21:27'` instead of 2024-07-13 like I would expect.

Reply:
It only uses expires at if no dag_expiry is set from dag.expiration_date above on l310. But you're saying that there's a problem in testing it?

Reply:
Gotcha, I didn't step through with the debugger, it might be that this works when the dag_expiry is set, but it doesn't work when it's not set, so we might just need to figure out how to test both cases.